CRM/Utils/System/Drupal8.php

```diff
@@ -300,19 +300,20 @@ public function url(
   ) {
     $query = html_entity_decode($query);
 
+    $config = CRM_Core_Config::singleton();
+    $base = $absolute ? $config->userFrameworkBaseURL : 'base:';
     $url = \Drupal\civicrm\CivicrmHelper::parseURL("{$path}?{$query}");
 
     // Not all links that CiviCRM generates are Drupal routes, so we use the weaker ::fromUri method.
     try {
-      $url = \Drupal\Core\Url::fromUri("base:{$url['path']}", [
+      $url = \Drupal\Core\Url::fromUri("{$base}{$url['path']}", [
```

Comment:
@mlutfy so that should be  ```\Drupal\Core\Url::fromUri("internal:{$url['path']}"``` right ?

Reply:
@monishdeb 

```
    $base = $absolute ? $config->userFrameworkBaseURL : 'internal:/';

    // Not all links that CiviCRM generates are Drupal routes, so we use the weaker ::fromUri method.
    try {
      $url = \Drupal\Core\Url::fromUri("{$base}{$url['path']}", [
      [...]
```

Before:

* CiviCRM generates non-localized URLs: `/civicrm/ajax/l10n-js/en_US?cid=131`

After:

* CiviCRM generates localized URLs: /en/civicrm/ajax/l10n-js/en_US?cid=131

nb: the `internal:/` prefix is important, otherwise URLs will be blank (Drupal8 URLs must start with a leading slash, but CiviCRM historally strips out the slash for other CMSes)

Reply:
I see thanks @mlutfy Will update the PR shortly